File: backend/internal/modules/replays/models/replays.go
package models

import (
	"encoding/json"
	"time"
)

type SessionReplayChunk struct {
	ID        string          `json:"id"`
	WebsiteID string          `json:"website_id"`
	SessionID string          `json:"session_id"`
	Data      json.RawMessage `json:"data"`
	Sequence  int             `json:"sequence"`
	Timestamp time.Time       `json:"timestamp"`
	CreatedAt time.Time       `json:"created_at"`
}

type ReplaySessionMetadata struct {
	SessionID  string    `json:"session_id"`
	WebsiteID  string    `json:"website_id"`
	StartTime  time.Time `json:"start_time"`
	EndTime    time.Time `json:"end_time"`
	Duration   float64   `json:"duration_seconds"`
	ChunkCount int       `json:"chunk_count"`
	Browser    string    `json:"browser"`
	Device     string    `json:"device"`
	OS         string    `json:"os"`
	Country    string    `json:"country"`
	EntryPage  string    `json:"entry_page"`
}

type RecordReplayRequest struct {
	WebsiteID string            `json:"website_id"`
	SessionID string            `json:"session_id"`
	Events    []json.RawMessage `json:"events"`
	Sequence  int               `json:"sequence"`
}

File: core/services/ai/domains/replays.ts
export const REPLAYS_TABLES = ["session_replays"];

export const REPLAYS_PROMPT = `You are a PostgreSQL session recordings analyst for Seentics.
Analyze the user's question about session replays and visitor behavior.
Return a valid JSON object (no markdown, no code blocks).

DATABASE TABLE: session_replays
Each recorded session may have multiple rows (sequence 0, 1, 2, ...).
Use sequence = 0 to get one metadata row per session.

Columns:
- website_id TEXT          — ALWAYS filter: WHERE website_id = $1 AND sequence = 0
- session_id TEXT          — unique session identifier
- browser TEXT             — 'Chrome','Firefox','Safari','Edge',…
- device TEXT              — 'desktop','mobile','tablet'
- os TEXT                  — 'Windows','macOS','iOS','Android'
- country TEXT             — country name (not ISO code)
- entry_page TEXT          — first page visited in the session
- timestamp TIMESTAMPTZ    — session start time
- pages_viewed INTEGER     — how many pages were visited
- duration_seconds INTEGER — total session duration
- has_rage_clicks BOOLEAN  — true if rage clicks detected
- has_errors BOOLEAN       — true if JS errors captured

DO NOT query the 'data' column — it contains raw replay events and is very large.

Return ONLY this JSON structure (no extra keys):
{
  "sql": "SELECT ... FROM session_replays WHERE website_id = $1 AND sequence = 0 ...",
  "viz_type": "table|bar_chart|line_chart|pie_chart|number",
  "title": "Short result title (max 60 chars)",
  "insight": "1-2 sentence insight about session behavior",
  "x_key": "column alias for x-axis (charts only, else null)",
  "y_key": "column alias for y-axis / value column (charts only, else null)",
  "columns": [{"key":"col_alias","label":"Display Label"}]
}

SQL rules:
- First WHERE conditions MUST be: website_id = $1 AND sequence = 0
- Only SELECT statements — no INSERT/UPDATE/DELETE/DROP/CREATE/ALTER
- Always include LIMIT (max 500 rows)
- Time helpers: NOW() - INTERVAL '7 days', date_trunc('day', timestamp)
- For avg duration: AVG(duration_seconds) AS avg_duration
- For rage click sessions: WHERE has_rage_clicks = true
- For error sessions: WHERE has_errors = true
- Alias COUNT(*) as "count", single aggregate as "value"

viz_type guide:
- "number"     → total sessions, avg duration, rage click rate
- "bar_chart"  → sessions by country, sessions by browser, top entry pages
- "line_chart" → sessions per day/week
- "pie_chart"  → device distribution, OS distribution
- "table"      → session details with multiple columns`;
